### DISCUSSION TITLE
How to edit river culvert/bridge so OsmAnd will navigate through

### DISCUSSION THREAD

**Discussion by lunkhub** [osmandapp/OsmAnd#19053] [Q&A]
Example location of LARGE culvert in river: 35.34422, -80.66594
Example boat navigation start, finish: 35.34239, -80.67029 (upstream) to 35.34396, -80.66504 (downstream of bridge/culvert)
Issue: OsmAnd navigates overland because of culvert?
How should map be changed to allow OsmAnd navigation along river?

I am aware of https://github.com/osmandapp/OsmAnd/issues/17752
but did not understand the recommendation there.
Thanks!

**Comment by lunkhub** [comment]
FYI, these two changes worked: Change culvert to tunnel with type=passage, and change section of road over culvert have structure=bridge. I do not know if only first change is sufficient.

**Comment by osmandlover** [comment]
I think a "culvert" is non-navigable by definition, they are usually pipes or boxes with very low headroom used for drainage. In the `/resources/routing/routing.xml` file on line 2544 culverts are correctly specified as non-navigable.

A "waterway=tunnel" with "boat=yes" is what you need to make it navigable. 

If you know of navigable waterway tunnels tagged "tunnel=culvert" then please change them in OSM to be _either_ a tunnel or the road to be a bridge. It sounds like you've done both, and that's probably not really correct, either the waterway is a tunnel or the road is a bridge, it is unlikely that both are true at the same time!

tunnel=passage is also not required to make it navigable but might help to make it clear, use your discretion there. 

boat=yes and optionally motorboat=yes should do the trick on pretty much any waterway!

**Reply by lunkhub** [reply]
@osmandlover, Thank you for your comments and suggestions. 

> It sounds like you've done both, and that's probably not really correct, either the waterway is a tunnel or the road is a bridge, it is unlikely that both are true at the same time!

I did enough web search to see too much discussion of possible definitions of bridges and tunnels. Definitions of tunnel using length, diameter, or depth underground or under-road, were not found. Definitions of road bridges using specific criteria also were not found. If you could point to official definitions, please do. Otherwise, I'll use "I know it when I see it". I feel like a "large" structure under a two lane road, made of metal and/or concrete, is a hybrid somewhere between obvious bridge or obvious tunnel. What harm comes from using both?

**Reply by osmandlover** [reply]
I think this is really a question for the OSM community rather than Osmand. Osmand is working as it should - it's considering a [tunnel=culvert](https://taginfo.openstreetmap.org/tags/tunnel=culvert) as non-navigable which is generally correct. 

The [OSM wiki](https://wiki.openstreetmap.org/wiki/Key:bridge) states:  
> Bridge or [tunnel](https://wiki.openstreetmap.org/wiki/Key:tunnel)? Where the lower feature is surrounded by earth then the lower feature should probably be tagged using tunnel=* instead. It is sometime, however, a matter of judgment as to whether to tag a particular situation using a bridge or a tunnel. But both options should not be used at the same time (a bridge tag for the upper way and a tunnel for the lower way for the same crossing).

And there's also an argument on [the help pages](https://help.openstreetmap.org/questions/5934/bridge-vs-tunnel) where someone suggested both, and that was met with some downvotes and this comment:
> Don't mark both a tunnel and bridge unless there really is a bridge crossing above a tunnel, which would be exceedingly rare. In nearly all cases it's one or the other, but not both.

But FWIW neither of these sources are gospel - the definitions are left to the discretion of the mapper and their interpretation of how best to represent the truth on the ground and following any conventions of their area. So I won't tell you that you're wrong! The best source is the map, and I can't say I've ever seen what you're describing - so the only harm that I can see from tagging both the waterway as a tunnel and the road as a bridge is that it may be inaccurate - it might appear to the renderer, or someone looking at the map, that the bridge is over a piece of ground, which has a tunnel in it.
